src/main/java/Domain/Role.java

```diff
@@ -0,0 +1,237 @@
+            throw new RoleException("Founder doesn't need other permissions.");
+        }
+
+        if(permissions.contains(Permission.FOUNDER) && appointedBy != null){
+            throw new RoleException("Founder is not appointed by anyone.");
+        }
+
+        if(permissions.contains(Permission.OWNER) && permissions.size() > 1){
+            throw new RoleException("Owner doesn't need other permissions.");
+        }
+        
+        _username = username;
+        _storeId = storeId;
+        _appointedBy = appointedBy;
+        _appointments = new HashSet<String>();
+        _permissions = permissions;
+    }
+
+    public boolean isFounder(){
+        return _permissions.contains(Permission.FOUNDER) && _appointedBy == null;
+    }
+
+    public boolean isOwner(){
+        return _permissions.contains(Permission.OWNER);
+    }
+
+    
+    /**
+     * Check if this role has a specific permission.
+     * @param p permission to check.
+     * @return boolean that represents if the role has this permission.
+     */
+    public boolean hasPermission(Permission p){
+        if(p == null){
+            return false;
+        }
+        return _permissions.contains(p);
+    }
+
+    /**
+     * Check if this role has at least one permission of the given set.
+     * @param permissions permission set.
+     * @return boolean that represents if the role has at least one permission from the set.
+     */
+    public boolean hasAtLeastOnePermission(Set<Permission> permissions){
+        if(permissions == null || permissions.isEmpty()){
+            return false;
+        }
+        permissions.retainAll(_permissions);
+        return !permissions.isEmpty();
+    }
+
+    /**
+     * Check if this role has at all permissions of the given set.
+     * @param permissions permission set.
+     * @return boolean that represents if the role has all permissions from the set.
+     */
+    public boolean hasAllPermissions(Set<Permission> permissions){
+        if(permissions == null || permissions.isEmpty()){
+            return false;
+        }
+        return !permissions.retainAll(_permissions);
+    }
+
+    //TODO: What happens if founder decides to change owner to regular manager?
+    /**
+     * Add permissions to this role.
+     *
+     * @param permission the permissions to add.
+     * @implNote If the role already has this permission its ok.
+     * @throws RoleException 
+     */
+    public void addPermissions(String username, Set<Permission> permissions) throws RoleException {
+        if(isFounder() || isOwner()){
+            throw new RoleException("Username is a founder of owner. No need to manage permissions.");
+        }
+        if(!username.equals(_appointedBy)){
+            throw new RoleException("Only the role that appointed this user can change permission.");
+        }
+        if(permissions.contains(Permission.OWNER) || permissions.contains(Permission.FOUNDER)){
```

Comment:
so how to creat a new OWNER? or we cant do that ?

Reply:
when creating an OWNER using the function in shop, he is granted the OWNER permission and this is the only one that he has and will have. I don't want users to give owner or founder permissions to regular manager. if we create an owner there is a separate function.

Reply:
I get that, but when you create an role for someonr- he will have a appinter on?

Reply:
when appointing someone he get the username of his appointer that cant be changed

Reply:
unless he is founder of course